docs/writing-rules/data-flow/taint-mode.md

```diff
@@ -7,28 +7,32 @@ description: >-
 
 # Taint tracking
 
-Semgrep supports intra-procedural [taint tracking](https://en.wikipedia.org/wiki/Taint_checking). This data-flow analysis tracks the flow of untrusted (**tainted**) data throughout a program. If tainted data are not transformed or checked accordingly (**sanitized**), taint analysis reports a finding whenever tainted data reach a vulnerable function (**sink**).
+Semgrep supports intra-procedural [taint tracking](https://en.wikipedia.org/wiki/Taint_checking). This data-flow analysis tracks the flow of untrusted (**tainted**) data throughout the body of a function or method. Tainted data originate from taint **sources**. If tainted data are not transformed or checked accordingly (**sanitized**), taint analysis reports a finding whenever tainted data reach a vulnerable function (what we call a **sink**). Tainted data flow from source to sink through **propagators** (assignments, function calls, etc).
 
 The following video provides a quick overview of taint mode:
 <iframe class="yt_embed" width="100%" height="432px" src="https://www.youtube.com/embed/6MxMhFPkZlU" frameborder="0" allowfullscreen></iframe>
 
+Getting started
+---------------
```

Comment:
Optional note - should we rather change these headers to `## Getting started` (The same applies to Propagators and other second-level headers in this article - they are easier to search for.) @IagoAbal

Reply:
Fine with me, I just think `--------` it's a bit more visual. There should also exist editor extensions that let you navigate through subsections. But I don't spend that much time in the docs so feel free to change it.